tests/components/smtp/test_notify.py

```diff
@@ -139,3 +68,83 @@ async def test_reload_notify(hass):
 
 def _get_fixtures_base_path():
     return path.dirname(path.dirname(path.dirname(__file__)))
+
+
+@pytest.fixture
+def message():
+    """Return MockSMTP object with test data."""
+    mailer = MockSMTP(
+        "localhost",
+        25,
+        5,
+        "test@test.com",
+        1,
+        "testuser",
+        "testpass",
+        ["recip1@example.com", "testrecip@test.com"],
+        "Home Assistant",
+        0,
+    )
+    yield mailer
+
+
+html = """
+        <!DOCTYPE html>
+        <html lang="en" xmlns="http://www.w3.org/1999/xhtml">
+            <head><meta charset="UTF-8"></head>
+            <body>
+              <div>
+                <h1>Intruder alert at apartment!!</h1>
+              </div>
+              <div>
+                <img alt="tests/testing_config/media/test.jpg" src="cid:tests/testing_config/media/test.jpg"/>
+              </div>
+            </body>
+        </html>"""
+
+email_data = [
+    ("Test msg"),
+    ("Test msg", {"images": ["tests/testing_config/media/test.jpg"]}),
+    ("Test msg", {"html": html, "images": ["tests/testing_config/media/test.jpg"]}),
+    ("Test msg", {"html": html, "images": ["test.jpg"]}),
+    ("Test msg", {"html": html, "images": ["tests/testing_config/media/test.pdf"]}),
+]
+
+
+@pytest.mark.parametrize(
+    "email_data",
+    [email_data[0], email_data[1], email_data[2], email_data[3], email_data[4]],
+    ids=[
+        "Tests when sending text message",
+        "Tests when sending text message and images.",
+        "Tests when sending text message, HTML Template and images.",
+        "Tests when image does not exist at mentioned location.",
+        "Tests when image type cannot be detected or is of wrong type.",
+    ],
+)
+def test_send_message(email_data, hass, message):
+    """Verify if we can send messages of all types correctly."""
+    sample_email = "<mock@mock>"
+    with patch("email.utils.make_msgid", return_value=sample_email):
+        if isinstance(email_data, tuple):
+            result = message.send_message(email_data[0], data=email_data[1])
+        elif isinstance(email_data, str):
+            result = message.send_message(email_data)
+
+        expected = (
+            '^Content-Type: text/plain; charset="us-ascii"\n'
+            "MIME-Version: 1.0\n"
+            "Content-Transfer-Encoding: 7bit\n"
+            "Subject: Home Assistant\n"
+            "To: recip1@example.com,testrecip@test.com\n"
+            "From: Home Assistant <test@test.com>\n"
+            "X-Mailer: Home Assistant\n"
+            "Date: [^\n]+\n"
+            "Message-Id: <[^@]+@[^>]+>\n"
+            "\n"
+            "Test msg$"
+        )
+        if isinstance(email_data, str):
+            assert re.search(expected, result)
```

Comment:
I would try to make parameters of this too, so we don't need to have any if checks.

Reply:
Sure let me know once we have that.

Reply:
I mean that we can extend the parameters in the pytest parametrization to avoid using checks in the test.

Reply:
If that's hard to do I would prefer to define multiple separate tests without pytest.parametrize which also would avoid checks inside the tests.